docs/source/mpi4py.util.pool.rst

```diff
@@ -0,0 +1,75 @@
+mpi4py.util.pool
+----------------
+
+.. module:: mpi4py.util.pool
+   :synopsis: :mod:`multiprocessing.pool` interface via :mod:`mpi4py.futures`.
+
+.. versionadded:: 4.0.0
+
+.. seealso::
+
+   This module intends to be a drop-in replacement for the
+   :mod:`multiprocessing.pool` interface from the Python standard library.
+   The :class:`~mpi4py.util.pool.Pool` class exposed here is implemented as a
+   thin wrapper around :class:`~mpi4py.futures.MPIPoolExecutor`.
```

Comment:
Do people need to care about the implementation detail?

Reply:
My point here was to somehow diminish the old multiprocessing API and direct readers to the more powerful and modern futures API. Perhaps I should have been more vehement about it, although I'm being in the note following this sentence. I believe a power user/developer - which are the ones I really care about - would appreciate a succinct and high level comment about implementation.

Reply:
Sounds good. Though you do also have a note right below advertising the `futures` API, which already serves your goal.